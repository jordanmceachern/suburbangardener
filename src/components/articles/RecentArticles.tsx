import React from "react";
import type { ArticleDisplayData } from "./ArticleDisplayCard";

interface RecentArticlesProps {
  articles: ArticleDisplayData[];
  className?: string;
}

export default function RecentArticles({
  articles,
  className = "",
}: RecentArticlesProps) {
  return (
    <div className={`recent-articles ${className}`}>
      <h2 className="text-xl font-semibold mb-4">Recent Articles</h2>
      <div className="space-y-3">
        {articles.slice(0, 5).map(article => (
          <div key={article.id} className="border-b border-gray-200 pb-2">
            <h3 className="font-medium text-sm">{article.title}</h3>
            <p className="text-xs text-gray-600">{article.publishedDate}</p>
          </div>
        ))}
      </div>
    </div>
  );
}
