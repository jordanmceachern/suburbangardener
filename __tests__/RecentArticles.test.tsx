import { render } from "@testing-library/react";
import RecentArticles from "@/components/articles/RecentArticles";

describe("RecentArticles Component", () => {
  const mockArticles = [
    {
      id: 1,
      title: "Recent Article 1",
      excerpt: "First recent article",
      slug: "recent-article-1",
      content: "Recent content 1",
      imageUrl: "/recent-image-1.jpg",
      tags: ["recent"],
      author: "Test Author",
      publishDate: "2024-01-01",
      readTime: 5,
    },
  ];

  it("should match DOM snapshot", () => {
    const { container } = render(<RecentArticles articles={mockArticles} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should maintain CSS classes for recent articles layout", () => {
    const { container } = render(<RecentArticles articles={mockArticles} />);

    // Check for recent articles classes
    const hasRecentClasses =
      container.innerHTML.includes("recent") ||
      container.innerHTML.includes("list") ||
      container.innerHTML.includes("space-") ||
      container.innerHTML.includes("divide-");

    expect(hasRecentClasses).toBe(true);
  });
});
