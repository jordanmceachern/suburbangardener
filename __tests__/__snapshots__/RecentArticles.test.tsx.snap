// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`RecentArticles Component should match DOM snapshot 1`] = `
<div
  class="recent-articles "
>
  <h2
    class="text-xl font-semibold mb-4"
  >
    Recent Articles
  </h2>
  <div
    class="space-y-3"
  >
    <div
      class="border-b border-gray-200 pb-2"
    >
      <h3
        class="font-medium text-sm"
      >
        Recent Article 1
      </h3>
      <p
        class="text-xs text-gray-600"
      />
    </div>
  </div>
</div>
`;
